### PR TITLE
Making JobsConfigurationManager easier to use with NuGet.Services.KeyVault

### PR DESCRIPTION
**Pull request by agr**
Added an overload of `JobsConfigurationManager.GetJobArgsDictionary` that accepts the `ISecretReaderFactory` from the NuGet.Services.KeyVault (the old one used it's own version of the `ISecretReaderFactory` which is slightly different from the one we have in NuGet.Services.KeyVault).

Moved common code between the old and a new methods into `ReadCommandLineArguments`

### CODE REVIEW

File: src/NuGet.Jobs.Common/Configuration/JobConfigurationManager.cs [hunk 1]
```diff
@@ -31,62 +32,28 @@ public static class JobConfigurationManager
...
+        /// <summary>
+        /// Parses the string[] of <c>args</c> passed into the job into a dictionary of string, string.
+        /// Expects the string[] to be set of pairs of argumentName and argumentValue, where, argumentName start with a hyphen
+        /// </summary>
+        /// <param name="logger">Logger to use for internal logging</param>
+        /// <param name="commandLineArgs">Arguments passed to the job via commandline or environment variable settings</param>
+        /// <param name="jobName">Jobname to be used to infer environment variable settings</param>
+        /// <param name="secretReaderFactory">Creates a secret reader.</param>
+        /// <returns>Returns a dictionary of arguments</returns>
+        public static IDictionary<string, string> GetJobArgsDictionary(ILogger logger, string[] commandLineArgs, string jobName, NuGet.Services.KeyVault.ISecretReaderFactory secretReaderFactory)
```
Comment (joelverhagen) [inline src/NuGet.Jobs.Common/Configuration/JobConfigurationManager.cs]: Do we need this full namespace?

Comment (joelverhagen) [reply src/NuGet.Jobs.Common/Configuration/JobConfigurationManager.cs]: Nevermind. I understand now.

Comment (joelverhagen) [reply src/NuGet.Jobs.Common/Configuration/JobConfigurationManager.cs]: Do we have plans to dedupe these two interfaces?

Comment (agr) [reply src/NuGet.Jobs.Common/Configuration/JobConfigurationManager.cs]: Added an [issue](https://github.com/NuGet/NuGetGallery/issues/4681) to drop it.

**Pull request merged**